modulemd/v2/tests/ModulemdTests/translation_helpers.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python3
+
+# This file is part of libmodulemd
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Fedora-License-Identifier: MIT
+# SPDX-2.0-License-Identifier: MIT
+# SPDX-3.0-License-Identifier: MIT
+#
+# This program is free software.
+# For more information on the license, see COPYING.
+# For more information on free software, see
+# <https://www.gnu.org/philosophy/free-sw.en.html>.
+
+import os
+import sys
+
+try:
+    import unittest
+    import gi
+    gi.require_version('Modulemd', '2.0')
+    from gi.repository import Modulemd
+except ImportError:
+    # Return error 77 to skip this test on platforms without the necessary
+    # python modules
+    sys.exit(77)
+
+from base import TestBase
+
+
+class TestTranslationHelpers(TestBase):
+
+    def test_translation_catalog_from_index(self):
+        # Create a Modulemd.ModuleIndex object containing the YAML stream
+        index = Modulemd.ModuleIndex.new()
+        yaml_path = "../test_data/f29.yaml"
+        ret, failures = index.update_from_file(yaml_path, True)
+        self.assertIsNotNone(index)
+
+        catalog = get_translation_catalog_from_index(index)
+        self.assertTrue(catalog)
+        self.assertEquals(len(catalog), 34)
+        _, err = catalog.check()
```

Comment:
This validates that the catalog is internally consistent, but it doesn't verify that it contains the set of strings we expect to have here.

What I'd like you to do here is iterate through the catalog with 
```python3
for msg in catalog:
```
I'd like you to get the `msg.string` and iterate through the list of `msg.locations` and confirm that the string matches at that location. This, plus the catalog count matching the expected number of them is probably enough to validate that it works.